core/build_playerglobal/src/lib.rs

```diff
@@ -15,6 +15,13 @@ use swf::Header;
 use swf::SwfStr;
 use swf::Tag;
 
+// The metadata name - all metadata in our .as files
+// should be of the form `[Ruffle(key1 = value1, key2 = value2)]`
+const RUFFLE_METADATA_NAME: &str = "Ruffle";
+// Indicates that we should generate a reference to an instance allocator
+// method (used as a metadata key with `Ruffle` metadata)
+const METADATA_INSTANCE_ALLOCATOR: &str = "InstanceAllocator";
```

Comment:
Is there any reason for not sticking with avmplus's `[native(cls=..., instance=..., construct=...)]` terminology? That way we will be able implement classes more closely to the original Flash Player, without needing to "translate" terms each time.

Reply:
We don't really have anything directly resembling the `cls=` and `instance=` design though.
That said, it'd be kinda nice to have stuff like `construct="none"` for non-instantiable classes like `Capabilities`, but that's unrelated to this PR or allocators.

Reply:
As @adrian17 mentioned, we already diverge from how avmplus implements things. I want to avoid giving the impression that our metadata has a directly equivalent meaning to avmplus metadata.

Reply:
This also has a benefit of giving us a bit more distance in case of copyright/licensing issues.